lib/http.js

```diff
@@ -23,11 +28,22 @@ const client = require('_http_client');
 const ClientRequest = exports.ClientRequest = client.ClientRequest;
 
 exports.request = function request(options, cb) {
+  if (typeof options === 'string') {
+    options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
+  } else if (options instanceof url.URL) {
```

Comment:
/cc @jasnell Is there a faster method we could use to check for this instead of using `instanceof`? Like checking for a symbol or something?